flink-runtime/src/main/java/org/apache/flink/runtime/state/LocalBackupAndRecoveryDirectoryProvider.java

```diff
@@ -48,7 +48,7 @@
  *
  * <p>
  */
-public interface LocalRecoveryDirectoryProvider extends Serializable {
+public interface LocalBackupAndRecoveryDirectoryProvider extends Serializable {
```

Comment:
This class is utilized by both LocalRecovery and LocalBackup. If we were to name it "LocalBackupDirectoryProvider," it would seem strange to create a LocalBackupDirectoryProvider when LocalBackup is disabled (but LocalRecovery is enabled). And naming it "LocalRecoveryDirectoryProvider" face the same dilemma.

How about we name it "LocalSnapshotDirectoryProvider"?

Reply:
The 'backup' means the copy of snapshot here, not do local backup in a narrow sense. local-recovery is omitted form 'recovery from a local backup', also needs a place providing the local backup. So I think local backup is a precise description for this usage. Well I agree to add 'Snapshot' here, 'SnapshotLocalBackupDirectoryProvider' or 'LocalSnapshotBackupDirectory' is fine with me. WDYT?

Reply:
maybe 'SnapshotLocalBackupDirectoryProvider' is too long, so LocalSnapshotDirectoryProvider is also fine with me